implementations/rust/ockam/ockam_api/src/cli_state/storage/projects_repository_sql.rs

```diff
@@ -44,22 +44,61 @@ impl ProjectsSqlxDatabase {
     pub async fn create() -> Result<Self> {
         Ok(Self::new(SqlxDatabase::in_memory("projects").await?))
     }
+
+    async fn is_shared_project(
+        &self,
+        project: &ProjectModel,
+        transaction: &mut AnyConnection,
+    ) -> Result<bool> {
+        // There are three possible scenarios:
+        // 1. The user enrolls and gets a project as an admin. There is an email from the "user"
+        //      table that matches an email in the project's "user_role" field that has role "admin".
+        // 2. Someone shares a project to the user via an invitation. There is an email from the "user"
+        //      table that matches an email in the project's "user_role" field that has role "service_user".
+        // 3. The user uses a ticket to enroll into a project. There is no email from the "user" table
+        //      that matches an email in the project's "user_role" field.
+        // This function returns true if we are in the second scenario; false otherwise.
+
+        // Get emails from user_roles that are not admins
+        let mut non_admin_emails = vec![];
```

Comment:
Maybe it is worth promoting this as method on `ProjectModel`?

Reply:
Not sure if we are going to need to do this outside of this. Maybe if I rewrite it using iterators would look better?


```rust
let non_admin_emails: Vec<String> = project
    .user_roles
    .iter()
    .filter(|user_role| user_role.role != RoleInShare::Admin)
    .map(|user_role| user_role.email.to_string())
    .collect();
```

Reply:
That works for me as well. I was just suggesting this because it's possible that we might need this in the future and won't think about coming back to this code to refactor.